frontend/src/component/ui/ViewChangeLogButton.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { Component, Fragment } from 'react';
+import { ChangeLogModal } from './ChangeLogModal';
+import { Update } from '../../entity/Update';
+
+interface Props {
+  onClick?: () => void;
+  changeLog?: Array<Update>;
+  closeModal: () => void;
+  shouldShowModal: boolean;
+  defaultVisibleLogs: number;
```

Comment:
I think defaultVisibleLogs should be pass as property for `ChangeLogModal.tsx` because it's a internal property of that component. This my favorite book on React app design: https://www.newline.co/fullstack-react/

Please at least build the first example  app from scratch to get some sense of good React app architecture. (I also did it myself: https://github.com/my-legacy-code/OnTrack)